# HG changeset patch
# User Sergey Mironov <grrwlf@gmail.com>
# Date 1381140490 -14400
#      Mon Oct 07 14:08:10 2013 +0400
# Node ID 04109cf0642a38d234a4efc7d31c76bc4bf22c02
# Parent  84b4f124ad75f38d24da2c331b0141cef2d0c29f
Add show instance for 'id' type

diff --git a/lib/ur/basis.urs b/lib/ur/basis.urs
--- a/lib/ur/basis.urs
+++ b/lib/ur/basis.urs
@@ -775,6 +775,7 @@
 type id
 val fresh : transaction id
 val giveFocus : id -> transaction unit
+val show_id : show id
 
 val dyn : ctx ::: {Unit} -> use ::: {Type} -> bind ::: {Type} -> [ctx ~ [Dyn]] => unit
           -> tag [Signal = signal (xml ([Dyn] ++ ctx) use bind)] ([Dyn] ++ ctx) [] use bind
diff --git a/src/monoize.sml b/src/monoize.sml
--- a/src/monoize.sml
+++ b/src/monoize.sml
@@ -1266,6 +1266,12 @@
             in
                 ((L'.EAbs ("s", s, s, (L'.ERel 0, loc)), loc), fm)
             end
+          | L.EFfi ("Basis", "show_id") =>
+            let
+                val s = (L'.TFfi ("Basis", "string"), loc)
+            in
+                ((L'.EAbs ("s", s, s, (L'.ERel 0, loc)), loc), fm)
+            end
           | L.EFfi ("Basis", "show_char") =>
             ((L'.EFfi ("Basis", "charToString"), loc), fm)
           | L.EFfi ("Basis", "show_bool") =>
